src/main/java/organice/logic/commands/DoneCommand.java

```diff
@@ -71,29 +81,68 @@ public boolean isValidDonorPatientPair(Nric firstNric, Nric secondNric, Model mo
         }
         if (model.hasPatient(patientNric) && model.hasDonor(donorNric)
                 && match(donor, patient)) {
-            //todo check if tasklist isempty
-            return true;
+            return donor.getStatus().isProcessing() && patient.getStatus().isProcessing();
```

Comment:
I believe that the proper logic is to check if the donor is under "processing" status to the specified patient. 

A possible invalid scenario is this:
Donor 1 and Patient 1 matches and is "processing", and Donor 2 and Patient 2 matches and is "processing". 
Donor 1 and Patient 2 would return true based on the code.

Reply:
You mean one of the pair is processing and another is not processing?
The pair which is not processing will return false since it havent been processed yet, it should not be done